File: 0303_Kostebelova_EK/lab3/src/matrix.cpp
#include "matrix.h"


Matrix::Matrix(int MATRIX_SIZE) {
    this->matrix = std::vector(MATRIX_SIZE, std::vector<int>(MATRIX_SIZE));
    for (int i = 0; i < MATRIX_SIZE; i++) {
        for (int j = 0; j < MATRIX_SIZE; j++) {
            this->matrix[i][j] = -10 + rand() % 21;
        }
    }
}

Matrix::Matrix(std::vector<std::vector<int>> matrixInput) {
    int size = matrixInput.size();
    this->matrix = std::vector(size, std::vector<int>(size));
    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {
            this->matrix[i][j] = matrixInput[i][j];
        }
    }
}

Matrix operator*(const Matrix &matrixA, const Matrix &matrixB) {
    int size = matrixA.matrix.size();
    auto resultMatrix = std::vector(size, std::vector<int>(size));
    int count;
    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {
            count = 0;
            for (int k = 0; k < size; k++) {
                count += (matrixA.matrix[i][k] * matrixB.matrix[k][j]);
            }
            resultMatrix[i][j] = count;
        }
    }
    return Matrix(resultMatrix);
}

std::ofstream& operator<<(std::ofstream& out, Matrix& matrixOutput) {
    for (int i = 0; i < matrixOutput.matrix.size(); i++) {
        for (int j = 0; j < matrixOutput.matrix[i].size(); j++) {
            out << matrixOutput.matrix[i][j] << " ";
        }
        out << std::endl;
    }
    out << std::endl;
    return out;
}

File: 0303_Tabakov_PL_lab3/src/lab3/Matrix.java
package lab3;

import java.util.Arrays;
import java.util.Random;

class Matrix {
    private final int[][] matrix;

    public Matrix(int MATRIX_SIZE) {
        matrix = new int[MATRIX_SIZE][MATRIX_SIZE];
        Random rand = new Random();
        for (int i = 0; i < MATRIX_SIZE; i++) {
            for (int j = 0; j < MATRIX_SIZE; j++) {
                matrix[i][j] = -10 + rand.nextInt(20);
                System.out.println(matrix[i][j]);
            }
        }
    }

    public Matrix(int[][] matrixInput) {
        int size = matrixInput.length;
        matrix = new int[size][size];
        for (int i = 0; i < size; i++) {
            System.arraycopy(matrixInput[i], 0, matrix[i], 0, size);
        }
    }

    public Matrix multiply(Matrix matrixB) {
        int size = this.matrix.length;
        int[][] resultMatrix = new int[size][size];
        int count;
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                count = 0;
                for (int k = 0; k < size; k++) {
                    count += (this.matrix[i][k] * matrixB.matrix[k][j]);
                }
                resultMatrix[i][j] = count;
            }
        }
        return new Matrix(resultMatrix);
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (int[] a:
             matrix) {
            sb.append(Arrays.toString(a)).append("\n");
        }

        return sb.toString();
    }
}